### PR TITLE
24 independent testing for extend ae specific

### PR DESCRIPTION
**Pull request by BrianLang**
First steps of the testing for extend ae specific

need advice on whether to implement same analysis/table in sas, or if i can simply write independent testing for what I assume should be correct output.

### CODE REVIEW

File: tests/testthat/test-independent-testing-extend_ae_specific.R [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  expect_error(extend_ae_specific_inference(test, ci = c(.1, .4)),
+    regex = "ci must be a single numeric."
+  )
+
+  expect_silent(extend_ae_specific_inference(test, ci = .95))
+})
+
+test_that("specific_inference ci same as SAS", {
+  ae_sp_inf_98 <- extend_ae_specific_inference(test, ci = .98)
```
Comment (LittleBeannie) [inline tests/testthat/test-independent-testing-extend_ae_specific.R:32]: May I know if the test is complete here?

File: tests/testthat/test-independent-testing-extend_ae_specific.R [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  ))
+})
+
+# extend_ae_specific_events:
+# The average number of events matches that from SAS
+
+test_that("specific_events: average number of events same as SAS", {
+  expect_silent(extend_ae_specific_events(test))
+
+  ae_sp_ev <- extend_ae_specific_events(test)
```
Comment (LittleBeannie) [inline tests/testthat/test-independent-testing-extend_ae_specific.R:69]: Same as above, is `ae_sp_ev` used afterward?

**Review by LittleBeannie — COMMENTED** [review]
Hi @BrianLang, thanks for proofreading the test of `prepare_ae_specific` and developing inputs checking in `extend_ae_specific`, `extend_ae_specific_events`. I only have 2 minor comments (see below).

**Comment by BrianLang** [comment]
Currently I have written tests for all things that are not sas related. Will need to discuss how to test the sas related points. `ae_sp_ev <- extend_ae_specific_events(test)` and `ae_sp_inf_98 <- extend_ae_specific_inference(test, ci = .98)` are preparation for comparison to independently tested outputs.

**Comment by fb-elong** [comment]
Shall we merge this PR and open another one for additional test? 

I feel we do not have to restricted on SAS output. If we notice difference, we can document and resolve them later.

**Comment by BrianLang** [comment]
I think this is ready for review @LittleBeannie.

File: tests/testthat/test-independent-testing-extend_ae_specific.R [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  ))
+
+  expect_true(all(c("dur", "dur_se") %in% names(ae_spec_dur)))
+})
+
+# extend_ae_specific_events:
+# The average number of events matches that from SAS
+
+test_that("specific_events: average number of events gives correct list entries", {
+  expect_silent(ae_spec_event <- extend_ae_specific_events(test))
```
Comment (LittleBeannie) [inline tests/testthat/test-independent-testing-extend_ae_specific.R:68]: Personally, I don't feel `expect_silent` is compensively enough as an independent test. The [test plan](https://github.com/Merck/metalite.ae/issues/24) covers the compression between `metalate.ae` and SAS.

File: tests/testthat/test-independent-testing-extend_ae_specific.R [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+  expect_true(all(c("events", "events_se") %in% names(ae_spec_event)))
```
Comment (LittleBeannie) [inline tests/testthat/test-independent-testing-extend_ae_specific.R:70]: Please test the numbers from `metalite.ae` is the same as that from SAS.

**Comment by elong0527** [comment]
As the existing test has all passed. Let's merge it first. For the rest item, let's have a separate PR.

**Review by elong0527 — APPROVED** [review]

**Pull request merged**